backend/fms_core/import_tool/sheet_data.py

```diff
@@ -0,0 +1,75 @@
+from django.core.exceptions import ValidationError
+from ._utils import data_row_ids_range, panda_values_to_str_list
+
+'''
+    SheetData objects
+
+    attributes (input): 
+        name, pandas dataframe, header row number, list of minimally required columns
+        
+    preview info from rows results (output): 
+        a dictionary with the sheet name, list of column headers, data sheet validity, 
+                              list of base_errors, list of rows_results
+'''
+
+class SheetData():
+    def __init__(self, name, dataframe, header_row_nb, minimally_required_columns):
+        self.name = name
+        self.dataframe = dataframe
+        self.header_row_nb = header_row_nb
+        self.minimally_required_columns = minimally_required_columns
+
+        self.dataframe.columns = self.dataframe.values[self.header_row_nb]
+
+        self.base_errors = []
+        self.is_valid = None
+
+        self.prepare_rows()
+
+
+    def prepare_rows(self):
+        self.rows = []
+        self.rows_results = []
+        for row_id in data_row_ids_range(self.header_row_nb + 1, self.dataframe):
+            row_data = self.dataframe.iloc[row_id]
+
+            print('row_data', row_data)
+
+            required_values = [row_data[key] for key in self.minimally_required_columns]
+            if any(list(map(lambda x: x is None, required_values))):
```

Comment:
All the lambda/map stacking are ugly and very poorly readable ... i would make functions out of them to at least convey the general idea of what is performed. Python list comprehensions are much more readable and can usually be used instead of lambda.

Reply:
Any/all are more pythonic and more efficient. They will break out the first time they encounter a value that does not respect the condition (as opposed to the list comprehension which will go through all the items)

Reply:
Any/all is the not the problem... the lambda is ... It's been introduced to many languages but it's basically a way to obfuscate your code. This one is not so bad i have seen worse though ... the idea being try the comprehensions to build your list and still use any if you want ... the efficiency gain is negligible here compared to the pain it is on the eye ... ;)

Reply:
Pain in the eye and in the variable names is always subjective ;) but ok I get it, you want me to keep any/all but change the lambda for a list comprehension